src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -58,26 +62,81 @@ private static Task createMarkedTask(Task task) {
     public CommandResult execute(Model model) throws CommandException {
```

Comment:
It might be better to abstract out the index checking process as well, to make the method clearer.